### PR TITLE
LMN-965 Add image gallery slideshow

### PR DESCRIPTION
**Pull request by marianeum**
Second part of the image gallery - the slideshow part which is used when there's no categories.

+ [ ] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [ ] Component `README.md`
+ [ ] Tests

If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)

### CODE REVIEW

File: app/src/androidTest/java/net/skyscanner/backpack/compose/imagegallery/BpkImageGalleryTest.kt [hunk 1]
```diff
@@ -28,4 +33,29 @@ class BpkImageGalleryTest : BpkSnapshotTest() {
...
     }
+
+    @Test
+    fun slideshow() {
+        recordModal { ImageGallerySlideshowStory() }
+    }
+
+    @Test
+    fun slideshow_landscape() {
+        val device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
```
Comment (marianeum) [inline app/src/androidTest/java/net/skyscanner/backpack/compose/imagegallery/BpkImageGalleryTest.kt:44]: we have a different layout for landscape - make sure we switch to landscape first

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/imagegallery/BpkImageGallery.kt [hunk 2]
```diff
@@ -18,60 +18,42 @@
...
-import net.skyscanner.backpack.compose.carousel.BpkCarouselState
-import net.skyscanner.backpack.compose.tokens.BpkSpacing
+import net.skyscanner.backpack.compose.imagegallery.internal.BpkImageGallerySlideshow
+import net.skyscanner.backpack.compose.modal.BpkModal
+import net.skyscanner.backpack.compose.modal.BpkModalState
+import net.skyscanner.backpack.compose.modal.rememberBpkModalState
+import net.skyscanner.backpack.compose.navigationbar.NavIcon
 
 @Composable
-fun BpkImageGalleryCarousel(
```
Comment (marianeum) [inline backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/imagegallery/BpkImageGallery.kt:39]: this moved to a different file

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/imagegallery/BpkImageGalleryCarousel.kt [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import net.skyscanner.backpack.compose.badge.BpkBadge
+import net.skyscanner.backpack.compose.badge.BpkBadgeType
+import net.skyscanner.backpack.compose.carousel.BpkCarousel
+import net.skyscanner.backpack.compose.carousel.BpkCarouselState
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+
+@Composable
+fun BpkImageGalleryCarousel(
```
Comment (marianeum) [inline backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/imagegallery/BpkImageGalleryCarousel.kt:39]: this is the existing carousel component that moved

File: app/build.gradle [hunk 4]
```diff
@@ -118,6 +118,7 @@ dependencies {
     androidTestImplementation libs.test.facebookSnapshotTestingCore
     androidTestImplementation libs.ktor.clientCore
     androidTestImplementation libs.ktor.clientCio
+    androidTestImplementation libs.test.uiautomator
```
Comment (marianeum) [inline app/build.gradle:121]: this is needed for a landscape test

**Review by alejandrorosas — APPROVED** [review]

**Pull request merged**